File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.cpp
using namespace std;

long long solution(int price, int money, int count)
{
    long long answer = -1;
    
    long long allPrice = price;
    int add = price;
    
    for(int i=2; i<=count; ++i)
    {
        price += add;
        allPrice += price;
    }
    if(money - allPrice >= 0)
        answer = 0;
    else
        answer = allPrice - money;
    
    return answer;
}

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.cs
using System;

class Solution
{
    public long solution(int price, int money, int count)
    {
        long totalCash = 0;
        for(int i = 1; i <= count; ++i)
            totalCash += i * price;
        if(totalCash >= money)
            return Math.Abs(money - totalCash);
        else
            return 0;
        
        return -1;
    }
}